### PR TITLE
Add action to resend the invitation email for a user

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Resend invitation email
Implemented an action to resend the invitation email for a user that hasnever been logged in yet.

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/4577

## Screenshots (if appropriate):

image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [x] Documentation ticket raised: <link> @mmattel required ? 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JanAckermann** [comment]
@phil-davis @JammingBen acceptance tests needed ?

**Comment by phil-davis** [comment]
> @phil-davis @JammingBen acceptance tests needed ?

We have existing test scenarios like: https://github.com/owncloud/enterprise/issues/4454
```
  Scenario Outline: user sets his own password after being created with an Email address only
    When the administrator creates a user with the name "<username>" and the email "guiusr1@owncloud" without a password using the webUI
    And the administrator logs out of the webUI
    And the user follows the password set link received by "guiusr1@owncloud" using the webUI
    And the user sets the password to "%regular%" and confirms with the same password using the webUI
    Then the user should be redirected to the login page
    And the email address "guiusr1@owncloud" should have received an email with the body containing
      """
      Password changed successfully
      """
    When the user logs in with username "<username>" and password "%regular%" using the webUI
    Then the user should be redirected to a webUI page with the title "Files - %productname%"
    Examples:
      | username | comment               |
      | guiusr1  | simple user-name      |
      | a@-+_.'b | complicated user-name |
```

We could add a scenario where the user "forgets" and the email needs to be sent again.

It will need some additions to the page object for the users page, so that the test can find where to click to re-send the email. Who should do that? (I can get someone to look at it tomorrow, if you like)

**Comment by JammingBen** [comment]
> It will need some additions to the page object for the users page, so that the test can find where to click to re-send the email. Who should do that? (I can get someone to look at it tomorrow, if you like)

We take care 👍

**Comment by ChrisEdS** [comment]
Will this action only be showed when the user was invited via email initially?

**Comment by JanAckermann** [comment]
> Will this action only be showed when the user was invited via email initially?
Nope

File: tests/acceptance/features/webUIAddUsers/addUsers.feature [hunk 1]
```diff
@@ -91,6 +91,25 @@ Feature: add users
       | guiusr1  | simple user-name      |
       | a@-+_.'b | complicated user-name |
 
+  @smokeTest @skipOnLDAP @skipOnOcV10.3
```
Comment (JanAckermann) [inline tests/acceptance/features/webUIAddUsers/addUsers.feature:94]: @phil-davis should we change the @skipOnOCV part?

Comment (phil-davis) [inline tests/acceptance/features/webUIAddUsers/addUsers.feature:94]:
```suggestion
  @smokeTest @skipOnLDAP @skipOnOcV10.6 @skipOnOcV10.7
```

Yes. This test will only pass on 10.8 (current master). So skip on recent releases that we care to test against (some oC10 apps test against latest and one earlier release)

File: tests/acceptance/features/webUIAddUsers/addUsers.feature [hunk 2]
```diff
@@ -91,6 +91,25 @@ Feature: add users
...
+  Scenario Outline: user sets his own password after being created with an Email address only and invitation link resend
+    When the administrator creates a user with the name "<username>" and the email "guiusr1@owncloud" without a password using the webUI
+    And the administrator resends invitation email for user with the name "<username>" using the webUI
+    And the administrator logs out of the webUI
+    And the user follows the password set link received by "guiusr1@owncloud" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUIAddUsers/addUsers.feature:99]: There should now be 2 emails waiting. Probably this scenario is going to pass even if the "resend click" does not work.
Maybe we can add a step that specifically follows the second link:
```suggestion
    And the user follows the second password set link received by "guiusr1@owncloud" using the webUI
```
The code for that step can specifically check that there are 2 emails, and then use the 2nd email.

**Comment by JanAckermann** [comment]
@phil-davis now we also cover, that the first link have an expired/invalid token. Thanks for the hint!

**Review by phil-davis — APPROVED** [review]
LGTM

**Comment by jvillafanez** [comment]
Maybe a designer should check the icon. It isn't crystal clear to me that you'll send an email. The color is also different from the trashbin icon.

**Review by jvillafanez — APPROVED** [review]
Code looks good

**Comment by JanAckermann** [comment]
@jvillafanez  there is a gray icon available, will have a look. In general NC has the same Icon, I am OK with it.

**Review by AlexAndBear — CHANGES_REQUESTED** [review]
- [x] Add title attribute for tooltip
- [x] Make icon look similar to delete

**Review by AlexAndBear — APPROVED** [review]

**Comment by mmattel** [comment]
Maybe stupid but why not in addition a occ command set where you can trigger to resend an invite to either a named or all never logged in like:
```
occ resend-invite:list
occ resend-invite:send --all
occ resend-invite:send <name>
```

**Comment by JammingBen** [comment]
> Maybe stupid but why not in addition a occ command set where you can trigger to resend an invite to either a named or all never logged in like:

I personally find it more intuitive via WebUI. The admin creates users via UI, therefore it makes sense to place this action there, too. These commands could be a nice addition though.

**Comment by JanAckermann** [comment]
@mmattel can you point out a real-world use-case for the command?

**Comment by phil-davis** [comment]
Note: IMO `occ user:add` does not currently have the ability to specify just the email address, and have an invitation email sent to the new user. It seems to always want a password. So maybe there are no admins currently using the command line to create users in this way.

**Pull request merged**